test/rest_api/test_storage_service.py

```diff
@@ -72,3 +72,26 @@ def test_storage_service_auto_compaction_tables(cql, this_dc, rest_api):
             assert resp.status_code == requests.codes.bad_request
 
     cql.execute(f"DROP KEYSPACE {keyspace}")
+
+def test_storage_service_keyspace_offstrategy_compaction(cql, this_dc, rest_api):
+    keyspace = new_keyspace(cql, this_dc)
+    with new_test_table(cql, keyspace, "a int, PRIMARY KEY (a)") as t0:
+        resp = rest_api.send("POST", f"storage_service/keyspace_offstrategy_compaction/{keyspace}")
+        resp.raise_for_status()
+
+    cql.execute(f"DROP KEYSPACE {keyspace}")
+
+def test_storage_service_keyspace_offstrategy_compaction_tables(cql, this_dc, rest_api):
+    keyspace = new_keyspace(cql, this_dc)
+    with new_test_table(cql, keyspace, "a int, PRIMARY KEY (a)") as t0:
+        with new_test_table(cql, keyspace, "a int, PRIMARY KEY (a)") as t1:
+            test_tables = [t0.split('.')[1], t1.split('.')[1]]
+
+            resp = rest_api.send("POST", f"storage_service/keyspace_offstrategy_compaction/{keyspace}", { "cf": f"{test_tables[0]},{test_tables[1]}" })
+            resp.raise_for_status()
+
+            # non-existing table
+            resp = rest_api.send("POST", f"storage_service/keyspace_offstrategy_compaction/{keyspace}", { "cf": f"{test_tables[0]},XXX" })
+            assert resp.status_code == requests.codes.bad_request
```

Comment:
It would be a little nicer to pump some sstables and see that it actually did offstrategy, then call it again and get a "did not do"

Reply:
we need to run repair to get offstrategy scheduled to run after the timeout so we can test the immediate triggering over the api, we plan to do that in a dtest (scylladb/scylla-dtest#2431)

Reply:
no change in v3